shared/lib_battery_dispatch_automatic_btm.cpp

```diff
@@ -804,15 +804,18 @@ void dispatch_automatic_behind_the_meter_t::check_power_restrictions(double& pow
     power = desiredCurrent * _Battery->V() * util::watt_to_kilowatt;
 }
 
-void dispatch_automatic_behind_the_meter_t::set_battery_power(FILE *p, const bool debug)
+void dispatch_automatic_behind_the_meter_t::set_battery_power(size_t idx, FILE *p, const bool debug)
 {
 	for (size_t i = 0; i != _P_target_use.size(); i++) {
+
+        double discharging_loss = _Battery->getDischargeLoss(idx + i, _dt_hour); // Units are kWac for AC connected batteries, and kWdc for DC connected
+
```

Comment:
See comment on lib_battery_powerflow - the battery is only covering discharging losses, so that's the only thing to plan for at this point in dispatch.